### PR TITLE
Rebased search indexer PR + batching

### PR DESCRIPTION
**Pull request by ray-kast**
- Added an AMQP queue config for search indexers.
- Scaffolded a consumer for search index jobs.
- Added foo upsert handler (FIXME) to indexer-search
- Got the search indexer actually running.
- metadata document index
- Refactored search dispatch code into a new module.

### CODE REVIEW

File: crates/indexer/src/http/metadata_json.rs [hunk 1]
```diff
@@ -528,3 +546,31 @@ pub async fn process<'a>(
...
+        .and_then(|u| {
+            let id = AssetIdentifier::new(&u);
+
+            proxy_url(client.proxy_args(), &id, Some(("width", "400")))
+                .map(|o| o.map(|u| u.to_string()))
+                .transpose()
+        })
+        .unwrap_or_else(|| Ok(image.to_owned()))?;
+
+    let document = MetadataDocument { image, raw };
```
Comment (dandlezzz) [inline crates/indexer/src/http/metadata_json.rs:567]: Is this enough data to create a link to the NFT? Wouldn't we need metadata_address?

**Review by dandlezzz — CHANGES_REQUESTED** [review]
Mostly great. Let's talk about what we are indexing and why.

File: crates/indexer/src/geyser/accounts/name_service.rs [hunk 2]
```diff
@@ -26,6 +26,17 @@ pub(crate) async fn process(
 
     let incoming_slot: i64 = slot.try_into()?;
 
+    let document = TwitterHandleDocument {
```
[the remark this answers is not in the document]
Comment (ray-kast) [reply crates/indexer/src/geyser/accounts/name_service.rs:29]: I don't think we're gonna be able to do that in a way that's performant or consistent.

File: crates/indexer/src/http/metadata_json.rs (hunk 1 above)
Comment (ray-kast) [reply crates/indexer/src/http/metadata_json.rs]: That's in the `id` field of the document we dispatch for upsert.

**Review by dandlezzz — APPROVED** [review]
approved.

**Pull request merged**